Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Login & Signup Flow/InitialLoginViewController.m

```diff
@@ -21,14 +21,33 @@ @implementation InitialLoginViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-    // Do any additional setup after loading the view.
+    self.usernameTextField.delegate = self;
+    self.passwordTextField.delegate = self;
+    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self.view action:@selector(endEditing:)];
+    [self.view addGestureRecognizer:tap];
 }
 
 - (IBAction)loginButtonTapped:(id)sender {
-    [self performSegueWithIdentifier:@"loginSegue" sender:nil];
+    NSString *inputUsername = self.usernameTextField.text;
+    NSString *inputPassword = self.passwordTextField.text;
+    if ((inputUsername.length == 0) || (inputPassword.length == 0)) {
+        UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Empty username or password." andMessage:@"Please input a username and password." okCompletion:nil cancelCompletion:nil];
+
+        [self presentViewController:alert animated:YES completion:nil];
+    } else {
+        [self performSegueWithIdentifier:@"loginSegue" sender:nil];
+    }
+}
 
+- (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    [textField resignFirstResponder];
+    if ([textField isEqual:self.passwordTextField]) {
```

Comment:
This is great, you can also make the password become the first responder when tapping next on the email field.